homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,370 @@
+"""
+Connect to a Samsung Printer via it's SyncThru
+ web interface and read data
+"""
+import logging
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_RESOURCE, STATE_UNKNOWN, CONF_HOST, CONF_NAME, CONF_FRIENDLY_NAME,
+    CONF_MONITORED_CONDITIONS)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+REQUIREMENTS = ['pysyncthru>=0.2.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'Samsung Printer'
+DEFAULT_MONITORED_CONDITIONS = [
+    'toner_black',
+    'toner_cyan',
+    'toner_magenta',
+    'toner_yellow',
+    'drum_black',
+    'drum_cyan',
+    'drum_magenta',
+    'drum_yellow',
+    'tray_1',
+    'tray_2',
+    'tray_3',
+    'tray_4',
+    'tray_5',
+    'output_tray_0',
+    'output_tray_1',
+    'output_tray_2',
+    'output_tray_3',
+    'output_tray_4',
+    'output_tray_5',
+]
+COLORS = [
+    'black',
+    'cyan',
+    'magenta',
+    'yellow'
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_RESOURCE): cv.url,
+    vol.Optional(
+        CONF_NAME,
+        default=DEFAULT_NAME
+    ): cv.string,
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS,
+        default=DEFAULT_MONITORED_CONDITIONS
+    ): vol.All(cv.ensure_list, [vol.In(DEFAULT_MONITORED_CONDITIONS)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the BLNET component"""
+
+    from pysyncthru import SyncThru, test_syncthru
+
+    if discovery_info is not None:
+        host = discovery_info.get(CONF_HOST)
+        name = discovery_info.get(CONF_NAME, DEFAULT_NAME)
+        # _LOGGER.info("Discovered a new Samsung Printer: %s" % discovery_info)
+        # Test if the discovered device actually is a syncthru printer
+        if test_syncthru(host) is False:
+            _LOGGER.error("No SyncThru Printer reached under given resource")
+            return False
+        # monitor all supported devices
```

Comment:
This comment is superfluous, the same applie to the comment below wrt monitored.